### PR TITLE
CMCL-1117: clipper namespace change to avoid conflicts

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR
https://jira.unity3d.com/browse/CMCL-1117

- Replaced clipper library (.dll) with source code (.cs). So we can track our modifications.
- Changed Clipper's `ClipperLib` namespace to `Cinemachine.ClipperLib` to avoid namespace conflict with others who include clipper. 

### Testing status

- [ ] Added an automated test
- [x] Passed all automated tests
- [x] Manually tested 

### Documentation status

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Technical risk
Checked performance difference between master and this. And they are the same (within tolerance, identical). Profiled on our sample scene and compared results in Profile Analyzer.

### CODE REVIEW

File: com.unity.cinemachine/Tests/Runtime/Cinemachine.Runtime.Tests.asmdef [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "allowUnsafeCode": false,
     "overrideReferences": true,
     "precompiledReferences": [
-        "nunit.framework.dll"
+        "nunit.framework.dll",
+        "cinemachine_clipper_library.dll"
```
Comment (gaborkb) [inline com.unity.cinemachine/Tests/Runtime/Cinemachine.Runtime.Tests.asmdef:19]: Confiner2DUnitTests depend on clipper, and they failed after not including it in some editor versions.

**Review by glabute — APPROVED** [review]
How did you change the namespace from ClipperLib to Cinemachine.ClipperLib?  I only see a "using" difference

**Review by glabute — CHANGES_REQUESTED** [review]
Need to source-control the customization to clipper

**Comment by gaborkb** [comment]
Yamato fails are due to splines package

**Review by glabute — APPROVED** [review]

**Pull request merged**